addons/analytic/models/analytic_account.py

```diff
@@ -93,6 +136,41 @@ class AccountAnalyticLine(models.Model):
     def _default_user(self):
         return self.env.context.get('user_id', self.env.user.id)
 
+
+
+    def _compute_amount_user_currency(self):
+        """ Compute the amount into the user's currency
+        """
+        context = dict(self._context or {})
+        user_currency_id = self.env.user.company_id.currency_id
+        ctx = context.copy()
+        for line in self:
+            ctx['date'] = line.date
+            company_currency_id = line.company_id.currency_id
+            amount = line.amount
+            if user_currency_id == company_currency_id:
+                line.amount_user_currency = amount
+            else:
+                line.amount_user_currency = company_currency_id.with_context(ctx).compute(amount, user_currency_id)
```

Comment:
Just call `.with_context(date=line.date)`, there's no need to do any context copy.